chat-android/src/main/java/com/ably/chat/Messages.kt

```diff
@@ -334,7 +339,8 @@ internal class DefaultMessages(
                 updateChannelSerialsAfterDiscontinuity(requireAttachSerial())
             }
         }
-        channel.on(channelStateListener)
+        @OptIn(InternalAPI::class)
+        channelWrapper.javaChannel.on(channelStateListener)
```

Comment:
Why we don't use `channel.on` directly since variable already exists

Reply:
we can,  but I intentionally don't use `channel` directly anywhere. This way it's harder to miss any unintentional usage